cmd/stellar-rpc/internal/methods/get_events.go

```diff
@@ -142,6 +128,23 @@ func (g *GetEventsRequest) Valid(maxLimit uint) error {
 		}
 	}
 
+	if g.Pagination != nil { //nolint:nestif
+		if g.Pagination.Cursor != nil && (g.StartLedger != 0 || g.EndLedger != 0) {
+			return errors.New("ledger ranges and cursor cannot both be set")
+		}
+		if g.Pagination.Limit > maxLimit {
+			return fmt.Errorf("limit must not exceed %d", maxLimit)
+		}
+	} else {
```

Comment:
I think this misses the case where there is a `limit` set but also `startLedger`/`endLedger`, which is why that check was separate in the previous implementation.

Reply:
Then, removing the else should be enough. I will do that

Reply:
Done. PTAL!